src/main/java/net/sf/jabref/exporter/SaveDatabaseAction.java

```diff
@@ -167,6 +169,66 @@ public void run() {
             panel.autoGenerateKeysBeforeSaving();
 
             if (FileBasedLock.waitForFileLock(panel.getBibDatabaseContext().getDatabaseFile(), 10)) {
+                // Check for external modifications to alleviate multiuser concurrency issue when near
+                // simultaneous saves occur to a shared database file:
+                if (panel.isUpdatedExternally()
```

Comment:
Could you please split your addition in multiple, smaller methods so that the code is easier to understand. Thanks.